controlplane/kubeadm/internal/controllers/controller.go

```diff
@@ -246,7 +246,7 @@ func patchKubeadmControlPlane(ctx context.Context, patchHelper *patch.Helper, kc
 }
 
 // reconcile handles KubeadmControlPlane reconciliation.
-func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane) (res ctrl.Result, reterr error) {
+func (r *KubeadmControlPlaneReconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster, kcp *controlplanev1.KubeadmControlPlane) (res ctrl.Result, reterr error) { //nolint:gocyclo
```

Comment:
Is this far off passing gocyclo? It would be  better to restructure a little maybe than to add this tag whenever a function goes over the limit.

Reply:
I think it just passed the limit - basing this on the observation that the gocyclo warning showed up after adding a new if branch to this function.